lib/Controller/AEnvironmentPageAwareController.php

```diff
@@ -67,12 +67,49 @@ public function loadSignRequestUuid(string $uuid): void {
 				'errors' => [$this->l10n->t('Invalid UUID')],
 			]), AppFrameworkHttp::STATUS_NOT_FOUND);
 		}
+	}
+
+	/**
+	 * @throws LibresignException
+	 */
+	public function validateSignRequestUuid(string $uuid): void {
+		if (!$this->signRequestEntity instanceof SignRequestEntity
+			|| !$this->fileEntity instanceof FileEntity) {
+			$this->loadSignRequestUuid($uuid);
+		}
 		$this->signFileService->validateSigner($uuid, $this->userSession->getUser());
 		$this->nextcloudFile = $this->signFileService->getNextcloudFile(
 			$this->fileEntity->getNodeId(),
 		);
 	}
 
+	/**
+	 * @throws LibresignException
+	 */
+	public function validateRenewSigner(string $uuid): void {
+		if (!$this->signRequestEntity instanceof SignRequestEntity
+			|| !$this->fileEntity instanceof FileEntity) {
+			$this->loadSignRequestUuid($uuid);
+		}
+		$this->signFileService->validateRenewSigner($uuid, $this->userSession->getUser());
+		$this->nextcloudFile = $this->signFileService->getNextcloudFile(
+			$this->fileEntity->getNodeId(),
+		);
+	}
+
+	/**
+	 * @throws LibresignException
+	 */
+	public function loadNextcloudFileFromSignRequestUuid(string $uuid): void {
+		if (!$this->signRequestEntity instanceof SignRequestEntity
+			|| !$this->fileEntity instanceof FileEntity) {
+			$this->loadSignRequestUuid($uuid);
+		}
```

Comment:
Repeated section of code in functions, we can extract this to another point or we can keep it for readability